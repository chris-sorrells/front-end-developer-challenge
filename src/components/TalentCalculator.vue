<template>
  <div class="talent-calculator">
    <talent-calculator-header>
      TitanStar Legends - Rune Mastery Loadout Talent Calculator 9000
    </talent-calculator-header>

    <div class="talent-tree-wrapper">
      <div class="talent-tree-container">
        <div class="talent-tree">
          <div class="title">Talent Path 1</div>
          <talent-path :runes="runes[0]" />
        </div>

        <div class="talent-tree">
          <div class="title">Talent Path 2</div>
          <talent-path :runes="runes[1]" />
        </div>
      </div>

      <talent-points />
    </div>
  </div>
</template>

<script>
import TalentPath from "@/components/TalentPath.vue";
import TalentPoints from "@/components/TalentPoints.vue";
import TalentCalculatorHeader from "@/components/TalentCalculatorHeader.vue";
import { mapGetters } from "vuex";

export default {
  components: {
    TalentPath,
    TalentPoints,
    TalentCalculatorHeader
  },

  computed: {
    ...mapGetters(["runes"])
  }
};
</script>

<style lang="scss">
.talent-calculator {
  border: 1px solid rgba(255, 255, 255, 0.15);
  margin: auto;
  padding: 20px 20px 50px 20px;
  background: url("../assets/talent-calc-bg.png") no-repeat;

  .talent-tree-wrapper {
    display: flex;

    .talent-tree-container {
      margin-right: 45px;

      .talent-tree {
        display: flex;
        margin-bottom: 45px;

        &:last-child {
          margin-bottom: 0;
        }

        .title {
          margin: auto 45px auto 0;
          text-transform: uppercase;
          font-size: 80%;
          font-weight: bold;
        }
      }
    }
  }
}
</style>
