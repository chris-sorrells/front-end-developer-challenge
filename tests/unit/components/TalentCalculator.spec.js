import { shallowMount, createLocalVue } from "@vue/test-utils";
import TalentCalculator from "@/components/TalentCalculator.vue";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("TalentCalculator.vue", () => {
  let getters;
  let store;

  beforeEach(() => {
    getters = {
      runes: () => [
        [
          { id: 1, type: "box", active: true, order: 1 },
          { id: 2, type: "silverware", active: false, order: 2 }
        ],
        [
          { id: 3, type: "boat", active: true, order: 1 },
          { id: 4, type: "goggles", active: true, order: 2 }
        ]
      ]
    };

    store = new Vuex.Store({
      getters
    });
  });

  it("renders", () => {
    const wrapper = shallowMount(TalentCalculator, {
      store,
      localVue
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
