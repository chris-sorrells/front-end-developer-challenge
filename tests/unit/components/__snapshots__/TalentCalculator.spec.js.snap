// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TalentCalculator.vue renders 1`] = `
<div
  class="talent-calculator"
>
  <talent-calculator-header-stub>
    
    TitanStar Legends - Rune Mastery Loadout Talent Calculator 9000
  
  </talent-calculator-header-stub>
   
  <div
    class="talent-tree-wrapper"
  >
    <div
      class="talent-tree-container"
    >
      <div
        class="talent-tree"
      >
        <div
          class="title"
        >
          Talent Path 1
        </div>
         
        <talent-path-stub
          runes="[object Object],[object Object]"
        />
      </div>
       
      <div
        class="talent-tree"
      >
        <div
          class="title"
        >
          Talent Path 2
        </div>
         
        <talent-path-stub
          runes="[object Object],[object Object]"
        />
      </div>
    </div>
     
    <talent-points-stub />
  </div>
</div>
`;
